src/com/palmergames/bukkit/towny/tasks/DailyTimerTask.java

```diff
@@ -206,23 +201,69 @@ protected void collectNationTaxes(Nation nation) throws EconomyException {
 				 * exists.
 				 * We are running in an Async thread so MUST verify all objects.
 				 */
-				if (townyUniverse.getDataSource().hasTown(town.getName())) {
+				if (universe.getDataSource().hasTown(town.getName())) {
 					if (town.isCapital() || !town.hasUpkeep())
 						continue;
-					if (!town.getAccount().payTo(nation.getTaxes(), nation, "Nation Tax")) {
-						localRemovedTowns.add(town.getName());		
-						town.removeNation();
+					if (town.getAccount().canPayFromHoldings(taxAmount)) {
+					// Town is able to pay the nation's tax.
+						town.getAccount().payTo(taxAmount, nation, "Nation Tax to " + nation.getName());
+						TownyMessaging.sendPrefixedTownMessage(town, Translation.of("msg_payed_nation_tax", TownyEconomyHandler.getFormattedBalance(taxAmount)));
 					} else {
-						TownyMessaging.sendPrefixedTownMessage(town, TownySettings.getPayedTownTaxMsg() + nation.getTaxes());
+					// Town is unable to pay the nation's tax.
+						if (!TownySettings.isTownBankruptcyEnabled() || (TownySettings.isTownBankruptcyEnabled() && !TownySettings.doBankruptTownsPayNationTax())) {

```

Comment:
This conditional can be reduced to `if (!TownySettings.isTownBankruptcyEnabled() || !TownySettings.doBankruptTownsPayNationTax())` Since you check first if bankruptcy is not enabled, then the only time `doBankruptTownsPayNationTax` will be checked is if that first conditional is false, and that means bankruptcy has to be enabled.